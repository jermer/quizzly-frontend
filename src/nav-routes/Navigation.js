import { useContext } from "react";
import { Link, NavLink } from "react-router-dom";
import UserContext from "../auth/UserContext";

import "./Navigation.css";

const Navigation = ({ logout }) => {

    const { currentUser } = useContext(UserContext);

    function loggedInNav() {
        return (
            <ul className="navbar-nav">
                <li className="nav-item me-2">
                    <NavLink className="nav-link" to="/discover">
                        <i className="fa-solid fa-compass me-1"></i>
                        Discover
                    </NavLink>
                </li>
                <li className="nav-item me-2">
                    <NavLink className="nav-link" to="/editor/library">
                        <i className="fa-solid fa-pen-to-square me-1"></i>
                        Library
                    </NavLink>
                </li>
                <li className="nav-item me-2">
                    <NavLink className="nav-link" to="/reports">
                        <i className="fa-solid fa-square-poll-vertical me-1"></i>
                        Reports
                    </NavLink>
                </li>

                <li className="nav-item dropdown me-2">
                    <a className="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
                        <i className="fa-solid fa-user"></i>
                    </a>
                    <ul className="dropdown-menu">
                        <li><span className="dropdown-item-text">
                            {currentUser.username}
                        </span></li>
                        <li><hr className="dropdown-divider" /></li>
                        <li className="dropdown-item">
                            <Link className="nav-link" to="/profile">
                                Edit Profile
                            </Link>
                        </li>
                        <li className="dropdown-item">
                            <Link className="nav-link" to="/" onClick={logout}>
                                <i className="fa-solid fa-right-from-bracket fa-flip-horizontal me-1"></i> Logout
                            </Link>
                        </li>

                    </ul>
                </li>

            </ul>
        );
    }

    function loggedOutNav() {
        return (
            <ul className="navbar-nav">
                <li className="nav-item">
                    <NavLink className="nav-link" to="/login">
                        Login
                    </NavLink>
                </li>
                <li className="nav-item">
                    <NavLink className="nav-link" to="/signup">
                        Sign Up
                    </NavLink>
                </li>
            </ul>
        );
    }

    return (
        <nav className="Navigation navbar navbar-expand-md">
            <Link className="navbar-brand ms-3 me-5 fw-bold" to="/">
                Quizzly
            </Link>
            {currentUser ? loggedInNav() : loggedOutNav()}
            {/* {loggedInNav()} */}
        </nav>
    );
}

export default Navigation;