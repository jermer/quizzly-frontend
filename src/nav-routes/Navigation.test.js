import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from './Navigation';

/**
 * 
 * NEED TO MOCK USER
 * 
 */

// smoke test
it('renders without crashing', () => {

    // render(
    //     <MemoryRouter>
    //         <Navigation />
    //     </MemoryRouter>
    // );
});

// snapshot test
it("matches snapshot", function () {

    // const { asFragment } = render(
    //     <MemoryRouter>
    //         <Navigation />
    //     </MemoryRouter>,
    // );
    // expect(asFragment()).toMatchSnapshot();
});
