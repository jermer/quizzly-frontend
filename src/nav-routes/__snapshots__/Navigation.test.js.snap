// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Quizzly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/discover"
        >
          Discover
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/creator/library"
        >
          Library
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/reports"
        >
          Reports
        </a>
      </li>
      <li
        class="nav-item dropdown mr-4"
      >
        <a
          aria-expanded="false"
          class="nav-link dropdown-toggle"
          data-bs-toggle="dropdown"
          href="#"
          role="button"
        >
          <i
            class="fa-solid fa-user"
          />
        </a>
        <ul
          class="dropdown-menu"
        >
          <li>
            <span
              class="dropdown-item-text"
            >
              testuser
            </span>
          </li>
          <li>
            <hr
              class="dropdown-divider"
            />
          </li>
          <li
            class="dropdown-item"
          >
            <a
              class="nav-link"
              href="/profile"
            >
              Edit Profile
            </a>
          </li>
          <li
            class="dropdown-item"
          >
            <a
              class="nav-link"
              href="/"
            >
              <i
                class="fa-solid fa-right-from-bracket fa-flip-horizontal me-1"
              />
               Logout
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Quizzly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/signup"
        >
          Sign Up
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
